src/App.jsx

```diff
@@ -14,33 +15,89 @@
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+export const App = () => {
+  const [goods, setGoods] = useState([...goodsFromServer]);
+  const [activeButton, setActiveButton] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  const funcSortAlphabetically = () => {
+    const sortedGoods = [...goods].sort();
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+    setGoods(isReversed ? sortedGoods.reverse() : sortedGoods);
+    setActiveButton('alphabetically');
+  };
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  const funcSortByLength = () => {
+    const sortedGoods = [...goods].sort((a, b) => {
+      if (a.length === b.length) {
+        return goodsFromServer.indexOf(a) - goodsFromServer.indexOf(b);
+      }
+
+      return a.length - b.length;
+    });
+
+    setGoods(isReversed ? sortedGoods.reverse() : sortedGoods);
+    setActiveButton('length');
+  };
+
+  const funcReverse = () => {
+    setGoods([...goods].reverse());
+    setIsReversed(!isReversed);
+  };
+
+  const funcReset = () => {
+    setGoods([...goodsFromServer]);
+    setActiveButton('');
+    setIsReversed(false);
+  };
+
+  const isOnStart = JSON.stringify(goodsFromServer) === JSON.stringify(goods);
```

Comment:
The `isOnStart` variable is used to determine if the current goods list matches the original list. This is useful for conditionally rendering the Reset button.